packages/aws-cdk-lib/aws-s3/lib/bucket.ts

```diff
@@ -651,6 +653,9 @@ export abstract class BucketBase extends Resource implements IBucket {
   public addToResourcePolicy(permission: iam.PolicyStatement): iam.AddToResourcePolicyResult {
     if (!this.policy && this.autoCreatePolicy) {
       this.policy = new BucketPolicy(this, 'Policy', { bucket: this });
+      if (this.removalPolicy === RemovalPolicy.RETAIN || this.removalPolicy === RemovalPolicy.RETAIN_ON_UPDATE_OR_DELETE) {
+        this.policy.applyRemovalPolicy(this.removalPolicy);
+      }
```

Comment:
Focusing only on resolving the issue in [this issue](https://github.com/aws/aws-cdk/issues/27125), it is only for `RETAIN` or `RETAIN_ON_UPDATE_OR_DELETE`.

- Since the default for `RemovalPolicy` is `DESTROY` in the S3 Bucket and BucketPolicies, there is little benefit to bother with `applyRemovalPolicy` even when `DESTROY` is used.
- If we did that, the scope of impact on existing implementations would be very large and we would need to change many more integ tests!